lib/puppet/util/network_device/netapp/facts.rb

```diff
@@ -54,9 +54,19 @@ def retrieve(host)
     # Set operatingsystem details if present
     if @facts['version'] then
       if @facts['version'] =~ /^NetApp Release (\d.\d(.\d)?\w*)/i
+        # Legacy facts
         @facts['operatingsystem'] = 'OnTAP'
-        @facts['os']['family'] = 'NetApp'
         @facts['operatingsystemrelease'] = $1
+        # New style facts
+        @facts['os'] = {}
+        @facts['os']['architecture'] = 'x86_64'
+        @facts['os']['family'] = 'OnTAP'
```

Comment:
is the change from 'NetApp' for osfamily intentional?

Reply:
Yeah I did that on purpose since the OS is called OnTAP and not NetApp, but if this breaks too much, then I can change it back to NetApp.

Reply:
If this is going to be changed, it's actually "ONTAP", not "OnTAP":
  https://www.netapp.com/us/products/data-management-software/ontap.aspx

Reply:
Good point! I've adjusted my commit so it now reads ONTAP everywhere instead of NetApp or OnTAP.